src/Stratis.Bitcoin.Features.Wallet/Interfaces/IWalletManager.cs

```diff
@@ -254,12 +254,11 @@ public interface IWalletManager
         /// </summary>
         /// <param name="walletName">The wallet name.</param>
         /// <param name="accountName">The account name.</param>
-        /// <param name="prevOutputTxTime">Previous OutputTxTime, used for pagination</param>
-        /// <param name="prevOutputIndex">Previous prevOutputIndex, used for pagination</param>
-        /// <param name="take">Number of records to Take</param>
         /// <param name="searchQuery">For now this can only be a transaction Id.</param>
+        /// <param name="limit">Limit the result set by this amount (primarily used for pagination).</param>
+        /// <param name="offset">Skip (offset) the result set by this amount (primarily used for pagination).</param>
         /// <returns>Collection of address history and transaction pairs.</returns>
-        IEnumerable<AccountHistory> GetHistory(string walletName, string accountName = null, long? prevOutputTxTime = null, int? prevOutputIndex = null, int? take = int.MaxValue, string searchQuery = null);
+        IEnumerable<AccountHistory> GetHistory(string walletName, string accountName = null, string searchQuery = null, int? limit = int.MaxValue, int? offset = 0);
```

Comment:
Using an offset is not 100% reliable due to the fact that the database can change between subsequent queries. After the change it will probably still work 99.999% of the time assuming rewinds/forks are not occurring all that frequently.

Reply:
Yeah, I'm not too concerned this. We have to remember that the history call is currently only used by the UI for the user's benefit. Unless this logic someday gets used for business critical results, I wouldn't worry about it too much.